dagger/node.nim

```diff
@@ -71,7 +70,7 @@ proc streamBlocks*(
     # to prevent slurping the entire dataset
     # since disk IO is blocking
     for c in blockManifest:
-      without blk =? (await node.blockStore.getBlock(c)):
+      without blk =? (await node.engine.retrieve(c)):
```

Comment:
In this PR, calling `node.engine.retrieve(c)` results in the engine first calling the block store manager, which checks if the block is available in any of the local block stores, e.g. the cache or file system store. If it’s available locally, then it will be returned. If it’s not available locally, the engine will attempt to get the block from the p2p network.

On one hand we have block stores which are resources managed by their nodes, and on the other we have the p2p network, i.e. the Dagger (Codex) Network. The main point being that a block store is a storage resource that logically sits “behind” a node. This represents a slight shift in terminology, where “block store” implies a storage resource that is (relatively) local to the node. Again that could be an in-memory cache or the local file system, but it could also be a NAS or S3 bucket or something like that.

By separating the concepts of block *store* and block exchange *network*, which in this PR involves dropping the idea/type `NetworkStore`, @emizzle and I were hoping to:

1. cut down on confusion when thinking about and discussing concepts related to “network”
2. remove a tight coupling of modules between `BlockExcEngine` and `NetworkStore`, where the only function of `NetworkStore` outside of the functionality covered by block store manager was to call the engine.